### ISSUE TITLE
How to get sparse feature importance from GNNExplainer 

### ISSUE THREAD

**Issue by BrownEyes1225** [dmlc/dgl#7232]
## ❓ Questions and Help

Hi to everyone!

I am using GNNExplainer[](https://doc.dgl.ai/en/latest/generated/dgl.nn.pytorch.explain.GNNExplainer.html#gnnexplainer) in DGL.

I have tried to increase the beta1 and beta 2 to larger values in order to get sparse feature importance vector, but no matter how large the two parameters are, the feature importance are not sparse.

Does ‘sparse’ mean the feature importance should have zeros? If so, why GNNExplainer fail to output sparse results?

Thanks a lot!

**Comment by BarclayII** [comment]
Both options only make most of the entries go close to zero (like 0.001) by entropy and mean regularization, but not exactly zero.  If you want a "truly" sparse explanation you could set a threshold and cut all the masks below it to zero.

**Comment by peizhou001** [comment]
Hi @BrownEyes1225, I am closing this issue assuming you are happy about our response. Feel free to follow up and reopen the issue if you have more questions with regard to our response.
